### PR TITLE
Remove Sesame OpenRDF libraries; rework stream-based serialization code

### PR DESCRIPTION
**Pull request by acoburn**
Resolves: https://jira.duraspace.org/browse/FCREPO-2242
Resolves: https://jira.duraspace.org/browse/FCREPO-2243

### CODE REVIEW

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 1]
```diff
@@ -2293,7 +2293,7 @@ public void testJsonLdProfile() throws IOException {
         final HttpPost method = postObjMethod();
         method.addHeader("Content-Type", "application/n3");
         final BasicHttpEntity entity = new BasicHttpEntity();
-        final String rdf = "<> <http://purl.org/dc/elements/1.1/title> \"this is a french title\"@fr ." +
+        final String rdf = "<> <http://purl.org/dc/elements/1.1/title> \"c'est un titre français\"@fr ." +
```
Comment (ajs6f) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java]: I think you mean `n'est un titre français`.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java [hunk 2]
```diff
@@ -113,75 +108,84 @@ public RdfStreamStreamingOutput(final RdfStream rdfStream, final Map<String, Str
...
+            final Writer w = new OutputStreamWriter(output, UTF_8);
+            final String profile = dataMediaType.getParameters().getOrDefault("profile", "");
+            JsonUtils.write(w, rdfToJsonLd(rdfStream, nsPrefixes, profile));
+
+        // For formats that can be block-streamed (n-triples, turtle)
+        } else if (format != null) {
+            LOGGER.debug("Stream-based serialization of {}", dataFormat.toString());
+            final StreamRDF stream = getWriterStream(output, format);
+            stream.start();
+            nsPrefixes.forEach((k, v) -> stream.prefix(k, v));
```
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: Isn't this `nsPrefixes.forEach(stream::prefix);`?

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java [hunk 3]
```diff
@@ -113,75 +108,84 @@ public RdfStreamStreamingOutput(final RdfStream rdfStream, final Map<String, Str
...
+                    o.getLiteralLanguage());
             } else {
-                return createLiteral(vfactory, object.getLiteralValue());
+                ds.addTriple(s.getURI(), p.getURI(), o.getURI());
             }
+        });
+        final JsonLdOptions opts = new JsonLdOptions();
+        opts.setCompactArrays(true);
+        final Object json = new JsonLdApi(opts).fromRDF(ds);
+        if (profile.equals("http://www.w3.org/ns/json-ld#compacted")) {
```
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: Should these format URIs be constants? Or are there already constants for them in our gigantic sack of dependencies?

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java (hunk 2 above)
Comment (acoburn) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: yes, thanks for catching that.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java (hunk 3 above)
Comment (acoburn) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: That would make sense -- the don't exist as constants in the JSON-LD library, but it doesn't stop us from defining them ourselves.

**Pull request merged**